sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -985,57 +985,6 @@ object GpuCast extends Arm {
     }
   }
 
-  def castStringToFloats(
-      input: ColumnVector,
-      ansiEnabled: Boolean,
-      dType: DType,
-      alreadySanitized: Boolean = false): ColumnVector = {
-    // 1. identify the nans
-    // 2. identify the floats. "null" and letters are not considered floats
-    // 3. if ansi is enabled we want to throw an exception if the string is neither float nor nan
-    // 4. convert everything that's not floats to null
-    // 5. set the indices where we originally had nans to Float.NaN
-    //
-    // NOTE Limitation: "1.7976931348623159E308" and "-1.7976931348623159E308" are not considered
-    // Inf even though Spark does
-
-    val NAN_REGEX = "^[nN][aA][nN]$"
-
-    val sanitized = if (alreadySanitized) {
-      input.incRefCount()
-    } else {
-      GpuCast.sanitizeStringToFloat(input, ansiEnabled)
```

Comment:
This was the only place where `sanitizeStringToFloat` was called, so that method should be removed.

Reply:
As discussed in the [JNI pull request](https://github.com/NVIDIA/spark-rapids-jni/pull/631), this kernel doesn't currently support hex values, so we won't be able to change the config unless that is addressed. That is an active discussion.

Reply:
The config removal is orthogonal to removing `GpuCast.sanitizeStringToFloat`.  The function is no longer referenced in the main code, regardless of config settings.  After this change it's only referenced by a test designed to test the function.